SalesforceCommerceCloud/pwa-kit
# Clean Up SDK Folder Structure

After changing how imports work by removing the `dist` path when importing, we can not go ahead and do some other small clean up tasks on the SDK folder structure. 

The reason for doing this is that when we initially started version 2.x we contained most of the changes in a single folder called `ssr` (server-side rending). So if you looked at the code base it was apparent that we did this after the fact. For example, we had multiple `utils` folders. 

So in this PR we take the opportunity to re-arrange things a little and remove some oddities. 

Below is a list of the changes: 

- Removed the `ssr` folder and promoted its contents up. 
- Changed the name of the `universal` folder to `react`
- Created a `hooks` folder for future user under react, and also created a `context` folder and placed the `device-context` in it.
- Moved utilities that were previously in the `ssr` folder into the root `utils` folder.
- Updated all references in the `pwa` and tests`
- Fixed the webpack import rewrite rules.

## Future benefits

1. The code is better organized to add some index files, so importing might look more like this `import {AppConfig} from 'pwa-kit-react-sdk/react/components`.
2. Setting up the `react` folder will give use a clear place to put all our future hooks, contexts, etc if we choose to do so.

 **GUS**: None
 **Linked PRs**: None

## TODO
- [x] Fix generated project failures

## How to test-drive this PR
- Ensure tests are passing.
- Generated a project and ensure it's working.
- Do some smoke testing on the PWA.
